### PR TITLE
Cleanup redundant null checks and use IsKind consistently

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn/issues/53218

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs [hunk 1]
```diff
@@ -153,7 +153,7 @@ private static bool IsControlBlock(SyntaxNode node)
                 }
             }
 
-            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentTokenParentParent != null && currentTokenParentParent is AccessorDeclarationSyntax)
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentTokenParentParent is AccessorDeclarationSyntax)
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs:156]: I'd love an analyzer for this...

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs [hunk 2]
```diff
@@ -180,7 +180,7 @@ private static bool IsControlBlock(SyntaxNode node)
             }
 
             // * { - in the Lambda context
-            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) && currentTokenParentParent != null &&
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) &&
                (currentTokenParentParent.IsKind(SyntaxKind.SimpleLambdaExpression) || currentTokenParentParent.IsKind(SyntaxKind.ParenthesizedLambdaExpression)))
```
Comment (sharwell) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs:184]: Do we also want to apply this simplification?

```suggestion
               currentTokenParentParent.IsKind(SyntaxKind.SimpleLambdaExpression, SyntaxKind.ParenthesizedLambdaExpression))
```

**Review by sharwell — APPROVED** [review]

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs (hunk 2 above)
Comment (mrahhal) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs:184]: If this PR closes #53218 I think it should. There are a lot of similar instances to be simplified in this file.

**Review by sharwell — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
thanks!

**Pull request merged**